### PR TITLE
Using function pointer contracts to model callbacks with the new contracts implementation

### PR DESCRIPTION
**Pull request by remi-delmas-3000**
The assumptions about callback functions consumed by HTTPClient_Send are now modelled using contracts.
Verifying this model requires this version of CBMC: https://github.com/diffblue/cbmc/pull/6887

### CODE REVIEW

File: test/cbmc/proofs/HTTPClient_Send-contracts/cbmc-viewer.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{ "expected-missing-functions":
```
Comment (aggarg) [inline test/cbmc/proofs/HTTPClient_Send-contracts/cbmc-viewer.json:1]: Do we need these log files?

Comment (remi-delmas-3000) [reply test/cbmc/proofs/HTTPClient_Send-contracts/cbmc-viewer.json:1]: Hello, this PR contains proofs relying on experimental features of the verificaiton tool CBMC, and as such is not yet meant to be merged.

I wasn't able to tag this PR as "experimental" or "do not merge".  Would you rather have me close it for now and reopen it later ?

Comment (aggarg) [reply test/cbmc/proofs/HTTPClient_Send-contracts/cbmc-viewer.json:1]: Thank you for your response. Yes, we can either mark this PR as "draft" or better would be to close and reopen when you are ready.

Comment (aggarg) [reply test/cbmc/proofs/HTTPClient_Send-contracts/cbmc-viewer.json:1]: As you mentioned above, I am closing this PR. Feel free to open a new PR when you are ready.

**Pull request closed without merging**